tests/plots/test_style.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import asdict
+
+import numpy as np
+
+from shap.plots import _style
+
+
+def test_default_style():
+    new_style = _style.load_default_style()
+    assert configs_are_equal(_style.STYLE, new_style)
+
+    new_style.secondary_color_negative = "black"
+    assert not configs_are_equal(_style.STYLE, new_style)
+
+
+def test_style_context():
+    custom_style = _style.StyleConfig(text_color="green")
+    assert _style.STYLE.text_color == "white"
+    with _style.style_context(custom_style):
+        assert _style.STYLE.text_color == "green"
+    assert _style.STYLE.text_color == "white"
+
+
+def test_style_overrides():
+    assert _style.STYLE.text_color == "white"
+    with _style.style_overrides(text_color="green"):
+        assert _style.STYLE.text_color == "green"
+    assert _style.STYLE.text_color == "white"
+
+
+# Helper functions to compare equality of config dataclasses
+
+
+def configs_are_equal(config1: _style.StyleConfig, config2: _style.StyleConfig):
+    d1 = asdict(config1)
+    d2 = asdict(config2)
+    return all(_values_are_equivalent(d1[key], d2[key]) for key in d1.keys())
```

Comment:
We might want to first check if d1 and d2 have the same keys, if not return False straight away. Otherwise it might blow up when it receives d2 without the `d1 key`, for example, and make it harder for us when debugging failing tests.

Reply:
Good shout. I suggested doing this as an `assert`, as presently we don't ever expect the keys to differ.